python/lsst/scarlet/lite/detect.py

```diff
@@ -0,0 +1,188 @@
+# This file is part of scarlet_lite.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import logging
+from typing import Sequence, cast
+
+import numpy as np
+from lsst.scarlet.lite.detect_pybind11 import Footprint  # type: ignore
```

Comment:
This is a problem anytime we monkey patch in the code (ie. use `@continue_class`). The problem is mypy thinks that `Footprint` isn't initialized yet, because it doesn't have a concept of monkey patching.